hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/LogFileIterator.scala

```diff
@@ -0,0 +1,323 @@
+        this.hasNextInternal
+      } else {
+        val projectedAvroRecord = requiredSchemaSafeAvroProjection(avroRecordOpt.get)
+        recordToLoad = deserialize(projectedAvroRecord)
+        true
+      }
+    }
+  }
+
+  override final def next(): InternalRow = recordToLoad
+
+  override def close(): Unit =
+    if (logScanner != null) {
+      try {
+        logScanner.close()
+      } finally {
+        logScanner = null
+      }
+    }
+}
+
+/**
+ * Provided w/ instance of [[HoodieMergeOnReadFileSplit]], provides an iterator over all of the records stored in
+ * Base file as well as all of the Delta Log files simply returning concatenation of these streams, while not
+ * performing any combination/merging of the records w/ the same primary keys (ie producing duplicates potentially)
+ */
+private class SkipMergeIterator(split: HoodieMergeOnReadFileSplit,
+                                baseFileReader: BaseFileReader,
+                                dataSchema: HoodieTableSchema,
+                                requiredSchema: HoodieTableSchema,
+                                tableState: HoodieTableState,
+                                config: Configuration)
+  extends LogFileIterator(split, dataSchema, requiredSchema, tableState, config) {
+
+  private val requiredSchemaUnsafeProjection = generateUnsafeProjection(baseFileReader.schema, structTypeSchema)
+
+  private val baseFileIterator = baseFileReader(split.dataFile.get)
+
+  override def hasNext: Boolean = {
+    if (baseFileIterator.hasNext) {
+      // No merge is required, simply load current row and project into required schema
+      recordToLoad = requiredSchemaUnsafeProjection(baseFileIterator.next())
+      true
+    } else {
+      super[LogFileIterator].hasNext
+    }
+  }
+}
+
+/**
+ * Provided w/ instance of [[HoodieMergeOnReadFileSplit]], provides an iterator over all of the records stored in
+ * a) Base file and all of the b) Delta Log files combining records with the same primary key from both of these
+ * streams
+ */
+class RecordMergingFileIterator(split: HoodieMergeOnReadFileSplit,
+                                baseFileReader: BaseFileReader,
+                                dataSchema: HoodieTableSchema,
+                                requiredSchema: HoodieTableSchema,
+                                tableState: HoodieTableState,
+                                config: Configuration)
+  extends LogFileIterator(split, dataSchema, requiredSchema, tableState, config) {
+
+  // NOTE: Record-merging iterator supports 2 modes of operation merging records bearing either
+  //        - Full table's schema
+  //        - Projected schema
+  //       As such, no particular schema could be assumed, and therefore we rely on the caller
+  //       to correspondingly set the scheme of the expected output of base-file reader
+  private val baseFileReaderAvroSchema = sparkAdapter.getAvroSchemaConverters.toAvroType(baseFileReader.schema, nullable = false, "record")
```

Comment:
Hi @YannByron 

Recently, we found an Avro schema issue which is caused by the wrong record name (detail here: https://github.com/apache/hudi/issues/7284).

May I ask if this line could cause the same problem? If so, we can discuss how to fix it in PR: https://github.com/apache/hudi/pull/7297

Reply:
The code is just moved from an other class. So I am not sure whether can work correctly in any cases.
But i have solved the very similar problem that caused by avro namespace: https://github.com/apache/hudi/commit/60b62fc25c0b7d34ff011a16c68b5d699c8e6bbc .

Reply:
Got it, thanks @YannByron 

Reviewed the code recently, turns out this Avro schema `baseFileReaderAvroSchema` is only used for `resolveNullableType` in `AvroSerializer`. It won't be involved in any serializer/deserializer process, so it's okay to use `"record"` as Avro schema name.